hudi-spark-datasource/hudi-spark-common/src/test/java/org/apache/hudi/testutils/SparkDatasetTestUtils.java

```diff
@@ -141,17 +141,18 @@ public static Dataset<Row> getRandomRowsWithKeys(SQLContext sqlContext,
    */
   public static Row getRandomValue(String partitionPath, boolean isError, String commitTime) {
     // order commit time, seq no, record key, partition path, file name
+    String recordKey = UUID.randomUUID().toString();
     Object[] values = new Object[9];
     values[0] = commitTime; //commit time
     if (!isError) {
       values[1] = ""; // commit seq no
     } else {
       values[1] = RANDOM.nextLong();
     }
-    values[2] = UUID.randomUUID().toString();
```

Comment:
The original logic would cause problems with record_key as precombine_key in tests because there was no unification of the _hoodi_record_key and the record_key